okhttp/src/main/java/okhttp3/OkHttpClient.java

```diff
@@ -216,6 +216,8 @@ public void apply(ConnectionSpec tlsConfiguration, SSLSocket sslSocket, boolean
   final int readTimeout;
   final int writeTimeout;
   final int pingInterval;
+  final EventListener eventListener;
```

Comment:
I’m unsure about using fields for both of these. Maybe we should have a ConstantFactory or something for the single value case, and when you set the listener that’s what you get?